inc/TRestAxionSolarHiddenPhotonFlux.h

```diff
@@ -0,0 +1,133 @@
+/*************************************************************************
+ * This file is part of the REST software framework.                     *
+ *                                                                       *
+ * Copyright (C) 2016 GIFNA/TREX (University of Zaragoza)                *
+ * For more information see http://gifna.unizar.es/trex                  *
+ *                                                                       *
+ * REST is free software: you can redistribute it and/or modify          *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * REST is distributed in the hope that it will be useful,               *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see http://www.gnu.org/licenses/.                             *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+#ifndef _TRestAxionSolarHiddenPhotonFlux
+#define _TRestAxionSolarHiddenPhotonFlux
+
+#include <TRestAxionSolarFlux.h>
+#include <TRestAxionSolarModel.h>
+
+//! A metadata class to load tabulated solar hidden photon fluxes. Mass and coupling set to 1.
+class TRestAxionSolarHiddenPhotonFlux : public TRestAxionSolarFlux {
+   private:
+    /// The hidden photon mass used for calculation
+    double HiddenPhotonMass;
+
+    /// The value of the kinetic mixing parameter used for calculation
+    double HiddenPhotonKineticMixing;
```

Comment:
This method is no needed. We should use the inherited data members inside `TRestAxionSolarFlux`.

```
std::string fCouplingType = ""; 
Double_t fCouplingStrength = 0;  //<
```

Inside the config file one has to define then:

```
<parameter name="couplingType" value="Chi" />
<parameter name="couplingStrength" value="1.e-10" />
```